### PR TITLE
Fix GpuSize

### PR DESCRIPTION
**Pull request by sperlingxx**
This pull request is to address #1974.

Signed-off-by: sperlingxx <lovedreamf@gmail.com>

### CODE REVIEW

**Comment by sperlingxx** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/collectionOperations.scala [hunk 1]
```diff
@@ -31,38 +31,17 @@ case class GpuSize(child: Expression, legacySizeOfNull: Boolean)
...
-      withResource(inputBase.getChildColumnView(-1)) { offset =>
-        withResource(offset.subVector(1)) { upBound =>
-          withResource(offset.subVector(0, offset.getRowCount.toInt - 1)) { lowBound =>
-            upBound.sub(lowBound)
-          }
-        }
-      }
-    }
 
     val nullScalar = if (legacySizeOfNull) {
```
Comment (revans2) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/collectionOperations.scala:35]: I think `countElements` already does the right thing for `legacySizeOfNull == false`. That should be the common case so why do the extra computation if it is not needed.

```
withResource(input.getBase.countElements()) { collectionSize =>
    if (legacySizeOfNull) {
        withResource(GpuScalar.from(-1)) { nullScalar =>
            withResource(input.getBase.isNull) { inputIsNull =>
              inputIsNull.ifElse(nullScalar, collectionSize)
            }
        }
    }  else {
      collectionSize.incRefCount()
    }
}
```

Comment (sperlingxx) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/collectionOperations.scala]: Fixed. Thx!

**Comment by sperlingxx** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**